csrc/scheduler/normalization_utils.cpp

```diff
@@ -698,11 +702,61 @@ std::optional<int64_t> getOptionalInnerOuterPersistentBufferBatches(
       inner_batch,
       ", batch_min = ",
       batch_min);
+
   const int64_t batch_max = getMaximumInnerOuterPersistentBufferBatch();
-  if (inner_batch <= batch_max) {
-    return inner_batch;
+
```

Comment:
I'd like to see some cleanups of the adjustments below. Not really sure how, though.

Reply:
I changed the algorithm to calculate batch and threads_per_block. In most cases, there is no change in the final batch and threads_per_block. In case with feature size of 1280 and 1600 the batch size is increased from 1 to 2, this change slightly increased the bandwidth about 10%.

Reply:
Looks better!